### DISCUSSION TITLE
`borg list`: error on non-existent path

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [borgbackup/borg#8273] [Ideas]
When listing archive contents with `borg list`, and specifying a non-existent path, the command exits with RC 0. I would expect an error to be raised.

MRE:

```
# List archive contents

Williams-MacBook-Air:~ williamedwards$ borg list /Users/williamedwards/testtest::testtest
Enter passphrase for key /Users/williamedwards/.config/borg/keys/testtest:
drwxr-xr-x williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir
-rw-r--r-- williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir/a
-rw-r--r-- williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir/b
-rw-r--r-- williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir/c

# Get existing path

Williams-MacBook-Air:~ williamedwards$ borg list /Users/williamedwards/testtest::testtest Users/williamedwards/randomdir
Enter passphrase for key /Users/williamedwards/.config/borg/keys/testtest:
drwxr-xr-x williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir
-rw-r--r-- williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir/a
-rw-r--r-- williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir/b
-rw-r--r-- williamedwards _appstore        0 Sun, 2024-07-07 15:46:46 Users/williamedwards/randomdir/c

# Get non-existent path

Williams-MacBook-Air:~ williamedwards$ borg list /Users/williamedwards/testtest::testtest Users/williamedwards/doesntexist
Enter passphrase for key /Users/williamedwards/.config/borg/keys/testtest:
Williams-MacBook-Air:~ williamedwards$ echo $?
0
```

TBF: whether a path doesn't exist _is_ easy to detect, because Borg simply returns no contents at all. While for an empty path, the path itself is returned.

**Comment by ThomasWaldmann** [comment]
What borg version(s)?
Is the repo local or remote?

borg list REPO?
or
borg list REPO::ARCHIVE?

**Reply by WilliamDEdwards** [reply]
> What borg version(s)?

1.2.4

AFAICS, this is not version-specific. 

> Is the repo local or remote?

Tested both.

**Comment by WilliamDEdwards** [comment]
Here's what I did in my Python library to work around this: https://github.com/CyberfusionIO/python3-cyberfusion-borg-support/commit/7b0c0d3fccfebcc0126956758f525cdfb548c6fe

**Comment by ThomasWaldmann** [comment]
Can you add a short reproducer / demo script to your top post that shows the problem?

**Reply by WilliamDEdwards** [reply]
Done.

**Comment by ThomasWaldmann** [comment]
IMHO, that's not a bug.

If a path pattern matches N pathes and borg output N pathes, it works correctly, including N == 0.

**Reply by WilliamDEdwards** [reply]
It _is_ a pattern, but when no typical pattern operators are used, I would expect it to be a literal match on the path - behaving as such, thus failing when that path doesn't exist.

Don't get me wrong: I do understand your point, and I don't perceive this to be a bug. Simply a suggestion from a user's perspective.

**Reply by ThomasWaldmann** [reply]
Somewhere in borg we have a warning (IIRC) if nothing matched.

Could be looked up and compared with this case.
